### PR TITLE
{bio}[GCC/4.8.4] DSRC-2.0rc, fastqz-1.5, fqzcomp-4.6, Quip-1.1.8, SCALCE-2.7, ZPAQ-7.00 (REVIEW)

### PR DESCRIPTION
**Pull request by fgeorgatos**
Signed-off-by: Fotis Georgatos <fotis@cern.ch>

### CODE REVIEW

**Comment by fgeorgatos** [comment]
Test report by @fgeorgatos
**SUCCESS**
Build succeeded for 6 out of 6
Linux debian 6.0.10, Intel(R) Xeon(R) CPU           L5640  @ 2.27GHz, Python 2.6.6
See https://gist.github.com/3c2ce0d7182c479d13a0 for a full test report.

**Comment by fgeorgatos** [comment]
@boegel: DSRC is currently provided via github only, so the Binary version has been preferred for now; if this is an issue, I could drop it from this PR. Otherwise, we just need one more review over here.

File: easybuild/easyconfigs/d/DSRC/DSRC-2.0rc-linux-64-bit.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+homepage = 'http://sun.aei.polsl.pl/dsrc'
+description = """DNA Sequence Reads Compression is an application designed for compression of data files
+ containing reads from DNA sequencing in FASTQ format. The amount of such files can be huge, e.g., a few
+ (or tens) of gigabytes, so a need for a robust data compression tool is clear. Usually universal compression
+ programs like gzip or bzip2 are used for this purpose, but it is obvious that a specialized tool can work better."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+# fi. http://sun.aei.polsl.pl/dsrc/download/2.0rc/dsrc
+sources = [name.lower()]
```
Comment (boegel) [inline easybuild/easyconfigs/d/DSRC/DSRC-2.0rc-linux-64-bit.eb]: versionless 'source', so include a checksum for it?

Comment (boegel) [inline easybuild/easyconfigs/d/DSRC/DSRC-2.0rc-linux-64-bit.eb]: and I'd prefer `%(namelower)s`

File: easybuild/easyconfigs/f/fastqz/fastqz-1.5-GCC-4.8.2.eb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+dependencies = [('ZPAQ', '6.57')]
+
+# eg. http://mattmahoney.net/dc/fastqz/fastqz15.cpp & http://mattmahoney.net/dc/fastqz/fapack.cpp
+minver = ''.join(version.split('.'))
+sources = [('fastqz%s.cpp' % minver, 'cp %s %(builddir)s/fastqz.cpp'), ('fapack.cpp', 'cp %s %(builddir)s')]
+source_urls = [homepage]
+
+premakeopts = 'g++ -O3 -msse2 -s -lpthread fastqz.cpp $EBROOTZPAQ/include/libzpaq.cpp -o fastqz'
+premakeopts += '&& g++ -O3 -s fapack.cpp -o fapack && /bin/true '
```
Comment (boegel) [inline easybuild/easyconfigs/f/fastqz/fastqz-1.5-GCC-4.8.2.eb]: `prebuildopts`, `premakeopts` is deprecated

File: easybuild/easyconfigs/s/SCALCE/SCALCE-2.7-GCC-4.8.2.eb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ handle other technologies (that generate base pair reads) if the read length is the same throughout the file."""
+
+toolchain = {'name': 'GCC', 'version': '4.8.2'}
+
+# eg. http://sourceforge.net/projects/scalce/files/2.7/scalce-v2.7-src.tar.gz/download
+sources = ['%(namelower)s-v%(version)s-src.tar.gz']
+source_urls = [SOURCEFORGE_SOURCE]
+
+preconfigopts = 'cd zlib-1.2.8 &&'
+makeopts = '-C zlib-1.2.8 && make all'
```
Comment (boegel) [inline easybuild/easyconfigs/s/SCALCE/SCALCE-2.7-GCC-4.8.2.eb]: `buildopts`

File: easybuild/easyconfigs/z/ZPAQ/ZPAQ-6.57-GCC-4.8.2.eb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ It supports input and output in the FASTQ and SAM/BAM formats,
+ compressing large datasets to as little as 15% of their original size."""
+
+toolchain = {'name': 'GCC', 'version': '4.8.2'}
+
+# eg. http://mattmahoney.net/dc/zpaq657.zip
+sources = ['%%(namelower)s%s.zip' % ''.join(version.split('.'))]
+source_urls = ['http://mattmahoney.net/dc/']
+
+makeopts = ' && pod2man zpaq.pod > zpaq.1'
```
Comment (boegel) [inline easybuild/easyconfigs/z/ZPAQ/ZPAQ-6.57-GCC-4.8.2.eb]: `buildopts`

File: easybuild/easyconfigs/f/fastqz/fastqz-1.5-GCC-4.8.2.eb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ It is one of the compressors described in the paper: Bonfield JK, Mahoney MV (2013) Compression of
+ FASTQ and SAM Format Sequencing Data. (mirror) PLoS ONE 8(3): e59190. doi:10.1371/journal.pone.0059190"""
+
+toolchain = {'name': 'GCC', 'version': '4.8.2'}
+
+dependencies = [('ZPAQ', '6.57')]
+
+# eg. http://mattmahoney.net/dc/fastqz/fastqz15.cpp & http://mattmahoney.net/dc/fastqz/fapack.cpp
+minver = ''.join(version.split('.'))
+sources = [('fastqz%s.cpp' % minver, 'cp %s %(builddir)s/fastqz.cpp'), ('fapack.cpp', 'cp %s %(builddir)s')]
```
Comment (boegel) [inline easybuild/easyconfigs/f/fastqz/fastqz-1.5-GCC-4.8.2.eb]: only source per line please?

File: easybuild/easyconfigs/f/fastqz/fastqz-1.5-GCC-4.8.2.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/f/fastqz/fastqz-1.5-GCC-4.8.2.eb]: don't hardcode optimisation flags, or even compiler names, use `... $CC $CFLAGS -s ...`

don't use the `/bin/true` trick, use the `CmdCp` easyblock instead

File: easybuild/easyconfigs/d/DSRC/DSRC-2.0rc-linux-64-bit.eb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+source_urls = [homepage + '/download/%(version)s/']
```
Comment (boegel) [inline easybuild/easyconfigs/d/DSRC/DSRC-2.0rc-linux-64-bit.eb]: I'd rather have the `homepage` bit copy-pasted than this string concatenation here...

File: easybuild/easyconfigs/f/fqzcomp/fqzcomp-4.6-GCC-4.8.2.eb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+description = """Fqzcomp is a basic fastq compressor, designed primarily for high performance.
+ Despite that it is comparable to bzip2 for compression levels."""
+
+toolchain = {'name': 'GCC', 'version': '4.8.2'}
+
+# eg. http://sourceforge.net/projects/fqzcomp/files/fqzcomp-4.6.tar.gz/download
+sources = [SOURCE_TAR_GZ]
+source_urls = [SOURCEFORGE_SOURCE]
+
+skipsteps = ['configure']
```
Comment (boegel) [inline easybuild/easyconfigs/f/fqzcomp/fqzcomp-4.6-GCC-4.8.2.eb]: this is the default with `MakeCp` already? so drop this?

File: easybuild/easyconfigs/s/SCALCE/SCALCE-2.7-GCC-4.8.2.eb [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+ to obtain better compression rate. SCALCE has been specifically designed for Illumina reads but it can
+ handle other technologies (that generate base pair reads) if the read length is the same throughout the file."""
+
+toolchain = {'name': 'GCC', 'version': '4.8.2'}
+
+# eg. http://sourceforge.net/projects/scalce/files/2.7/scalce-v2.7-src.tar.gz/download
+sources = ['%(namelower)s-v%(version)s-src.tar.gz']
+source_urls = [SOURCEFORGE_SOURCE]
+
+preconfigopts = 'cd zlib-1.2.8 &&'
```
Comment (boegel) [inline easybuild/easyconfigs/s/SCALCE/SCALCE-2.7-GCC-4.8.2.eb]: hmm, wouldn't it be better to properly listed `zlib` as a dependency?

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
@fgeorgatos: some concerns w.r.t. deprecation, Jenkins will check that for you ;)

File: easybuild/easyconfigs/f/fqzcomp/fqzcomp-4.6-GCC-4.8.2.eb (hunk 7 above)
Comment (fgeorgatos) [inline easybuild/easyconfigs/f/fqzcomp/fqzcomp-4.6-GCC-4.8.2.eb]: @boegel: agreed, fixed this and most other bits;
negotiation pending as regards a SCALCE dep...

File: easybuild/easyconfigs/s/SCALCE/SCALCE-2.7-GCC-4.8.2.eb (hunk 8 above)
Comment (fgeorgatos) [inline easybuild/easyconfigs/s/SCALCE/SCALCE-2.7-GCC-4.8.2.eb]: @boegel: dunno; 
SCALCE carries with it its own version of zlib so I made the assumption there must be a cause;
however, it's only an assumption in the name of safety...

File: easybuild/easyconfigs/z/ZPAQ/ZPAQ-6.57-GCC-4.8.2.eb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+sources = ['%%(namelower)s%s.zip' % ''.join(version.split('.'))]
+source_urls = ['http://mattmahoney.net/dc/']
+
+buildopts = ' && pod2man zpaq.pod > zpaq.1'
+
+files_to_copy = [(["zpaq"], 'bin'), (["zpaq.1"], 'share/man'), (["libzpaq.h", "libzpaq.cpp"], 'include')]
+
+sanity_check_paths = {
+    'files': ["bin/zpaq", "share/man/zpaq.1"], 
+    'dirs': []
```
Comment (boegel) [inline easybuild/easyconfigs/z/ZPAQ/ZPAQ-6.57-GCC-4.8.2.eb]: also check the include bits?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
Linux SL 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2fad5ffcdcb915a58f65 for a full test report.

File: easybuild/easyconfigs/s/SCALCE/SCALCE-2.7-GCC-4.8.2.eb (hunk 8 above)
Comment (boegel) [inline easybuild/easyconfigs/s/SCALCE/SCALCE-2.7-GCC-4.8.2.eb]: The resulting installation doesn't provide zlib, or require the included zlib to be around, so it's fine I guess.

However, funnily enough, it also depends on `bzip2`:

```
$ ldd /gpfs/scratch/projects/project_gpilot/vsc40023/software/SCALCE/2.7-GCC-4.8.2/bin/scalce 
        linux-vdso.so.1 =>  (0x00007fffbbbff000)
        libbz2.so.1 => /lib64/libbz2.so.1 (0x00002b3b9a8ef000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00002b3b9ab00000)
        libstdc++.so.6 => /usr/lib64/libstdc++.so.6 (0x00002b3b9ad1e000)
        libm.so.6 => /lib64/libm.so.6 (0x00002b3b9b024000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00002b3b9b2a8000)
        libc.so.6 => /lib64/libc.so.6 (0x00002b3b9b4bf000)
        /lib64/ld-linux-x86-64.so.2 (0x00002b3b9a6cd000)
```

so, we should include a dependency on `bzip2` in here?

File: easybuild/easyconfigs/z/ZPAQ/ZPAQ-6.57-GCC-4.8.2.eb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+ It supports input and output in the FASTQ and SAM/BAM formats,
+ compressing large datasets to as little as 15% of their original size."""
+
+toolchain = {'name': 'GCC', 'version': '4.8.2'}
+
+# eg. http://mattmahoney.net/dc/zpaq657.zip
+sources = ['%%(namelower)s%s.zip' % ''.join(version.split('.'))]
+source_urls = ['http://mattmahoney.net/dc/']
+
+buildopts = ' && pod2man zpaq.pod > zpaq.1'
```
Comment (boegel) [inline easybuild/easyconfigs/z/ZPAQ/ZPAQ-6.57-GCC-4.8.2.eb]: I suspect this introduces some kind of build dependency, no?

**Comment by fgeorgatos** [comment]
Test report by @fgeorgatos
**SUCCESS**
Build succeeded for 7 out of 7
Linux debian 6.0.10, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.6.6
See https://gist.github.com/e64ec34a0a9515e5b306 for a full test report.

**Comment by fgeorgatos** [comment]
OK, here it is; bug fixed and retested. Just make your judicium on the pod2man bit.

**Comment by fgeorgatos** [comment]
Test report by @fgeorgatos
**SUCCESS**
Build succeeded for 7 out of 7
Linux debian 6.0.10, Intel(R) Xeon(R) CPU E5-2660 0 @ 2.20GHz, Python 2.6.6
See https://gist.github.com/6d7299ff33e107c99409 for a full test report.

File: easybuild/easyconfigs/z/ZPAQ/ZPAQ-7.00-GCC-4.8.2.eb [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+homepage = 'http://mattmahoney.net/dc/zpaq.html'
+description = """zpaq is a free and open source (GPL v3) incremental,
+ journaling command-line archiver for Windows, Linux and Mac OS/X"""
+
+toolchain = {'name': 'GCC', 'version': '4.8.2'}
+
+sources = ['%%(namelower)s%s.zip' % ''.join(version.split('.'))]
+source_urls = ['http://mattmahoney.net/dc/']
+
+buildopts = " && pod2man zpaq.pod > zpaq.1"
```
Comment (boegel) [inline easybuild/easyconfigs/z/ZPAQ/ZPAQ-7.00-GCC-4.8.2.eb]: Either include a build dependency for `pod2man`, or remove this all together...

As is, this is a liability for breaking this build...

Comment (boegel) [inline easybuild/easyconfigs/z/ZPAQ/ZPAQ-7.00-GCC-4.8.2.eb]: This is interesting:

```
$ rpm -q --whatprovides `which pod2man`
perl-5.10.1-136.el6.x86_64
```

and indeed:

```
$ module load Perl/5.16.3-goolf-1.4.10-bare
$ which pod2man
/gpfs/scratch/projects/project_gpilot/vsc40023/software/Perl/5.16.3-goolf-1.4.10-bare/bin/pod2man
```

So, adding a Perl build on top of `GCC/4.8.2` in here as a build dependency would do it!

Comment (boegel) [inline easybuild/easyconfigs/z/ZPAQ/ZPAQ-7.00-GCC-4.8.2.eb]: see https://github.com/fgeorgatos/easybuild-easyconfigs/pull/35

**Comment by fgeorgatos** [comment]
@boegel: 
I have some doubts if we should add the most recent Perl as a builddep for a manpage; 
however, that should not be a blocker for the whole PR, so now it's MERGED!

**Comment by boegel** [comment]
@fgeorgatos: the only alternative is removed the use of `pod2man`... And it's not unlikely that these bioinfo people will like having Perl around anyway ;-)

**Comment by boegel** [comment]
@fgeorgatos: this needs https://github.com/fgeorgatos/easybuild-easyconfigs/issues/36...

**Comment by fgeorgatos** [comment]
@boegel: OK, merged; I think we (I) have entered bikesheding mode, so I just let go ;-)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux SL 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/b1e06431599384e6a510 for a full test report.

**Comment by boegel** [comment]
Looking great now, going in. Thanks @fgeorgatos!

**Comment by fgeorgatos** [comment]
Pointers for future work under the same thread (if anybody gets interested in this anytime soon):

* [ ] LEON http://www.biomedcentral.com/1471-2105/16/288/
* [ ] DeeZ http://sfu-compbio.github.io/deez/

also, keep an eye on: http://ga4gh.org/#/fileformats-team

**Pull request merged**